src/Symfony/Component/HttpClient/Tests/CurlHttpClientTest.php

```diff
@@ -28,7 +38,81 @@ protected function getHttpClient(string $testCase): HttpClientInterface
+        $logger = new TestLogger();
+        $client->setLogger($logger);
+
+        $responseAsArray = $client->request('GET', 'https://127.0.0.1:3000/json', [
+            'headers' => [
+                'Preload' => '/documents/*/id',
+            ],
+        ])->toArray();
+
+        foreach ($responseAsArray['documents'] as $document) {
+            $client->request('GET', 'https://127.0.0.1:3000'.$document['id'])->toArray();
+        }
+
+        $client->reset();
+
+        $expected = [
+            'Request: "GET https://127.0.0.1:3000/json"',
+            'Queueing pushed response: "https://127.0.0.1:3000/json/1"',
+            'Queueing pushed response: "https://127.0.0.1:3000/json/2"',
+            'Queueing pushed response: "https://127.0.0.1:3000/json/3"',
+            'Response: "200 https://127.0.0.1:3000/json"',
+            'Accepting pushed response: "GET https://127.0.0.1:3000/json/1"',
+            'Response: "200 https://127.0.0.1:3000/json/1"',
+            'Accepting pushed response: "GET https://127.0.0.1:3000/json/2"',
+            'Response: "200 https://127.0.0.1:3000/json/2"',
+            'Accepting pushed response: "GET https://127.0.0.1:3000/json/3"',
+            'Response: "200 https://127.0.0.1:3000/json/3"',
+        ];
+        $this->assertSame($expected, $logger->logs);
+    }
+
+    /**
+     * @requires PHP 7.2.17
+     */
+    public function testHttp2PushVulcainWithUnusedResponse()
+    {
+        $client = $this->getVulcainClient();
+        $logger = new TestLogger();
+        $client->setLogger($logger);
+
+        $responseAsArray = $client->request('GET', 'https://127.0.0.1:3000/json', [
+            'headers' => [
+                'Preload' => '/documents/*/id',
+            ],
+        ])->toArray();
+
+        $i = 0;
+        foreach ($responseAsArray['documents'] as $document) {
+            $client->request('GET', 'https://127.0.0.1:3000'.$document['id'])->toArray();
+            if (++$i >= 2) {
+                break;
+            }
+        }
+
+        $client->reset();
+
+        $expected = [
+            'Request: "GET https://127.0.0.1:3000/json"',
+            'Queueing pushed response: "https://127.0.0.1:3000/json/1"',
+            'Queueing pushed response: "https://127.0.0.1:3000/json/2"',
+            'Queueing pushed response: "https://127.0.0.1:3000/json/3"',
+            'Response: "200 https://127.0.0.1:3000/json"',
+            'Accepting pushed response: "GET https://127.0.0.1:3000/json/1"',
+            'Response: "200 https://127.0.0.1:3000/json/1"',
+            'Accepting pushed response: "GET https://127.0.0.1:3000/json/2"',
+            'Response: "200 https://127.0.0.1:3000/json/2"',
+            'Unused pushed response: "https://127.0.0.1:3000/json/3"',
+        ];
+        $this->assertSame($expected, $logger->logs);
+    }
+
+    private function getVulcainClient(): CurlHttpClient
     {
         if (\PHP_VERSION_ID >= 70300 && \PHP_VERSION_ID < 70304) {
```

Comment:
Maybe that this test should also be skipped if `vulcain` isn't available?

Reply:
I don't know. An error will pop if it is not installed.

Reply:
We'll see the day someone hits this :)